File: day01/a.exs
File.read!("./input.txt")
|> String.split("\n", trim: true)
|> Enum.map(fn l ->
    l
    |> then(&Regex.scan(~r/\d/, &1))
    |> List.flatten()
    |> then(&(Enum.at(&1, 0) <> Enum.at(&1, -1)))
    |> String.to_integer()
    end)
|> Enum.sum()
|> IO.inspect()

File: day01/a.ts
const res = (await Deno.readTextFile("./input.txt"))
  .trim()
  .split("\n")
  .map((line: string) => {
    const nums = line.match(/\d/g) as Array<string>;
    return nums?.at(0)! + nums?.at(-1)!;
  })
  .reduce((a: number, c: string) => a + parseInt(c), 0);
console.log(res);
